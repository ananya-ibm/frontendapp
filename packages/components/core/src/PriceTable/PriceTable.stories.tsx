/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { PriceTable } from './PriceTable';

export default {
  title: 'Components/Core/PriceTable',
  component: PriceTable
};

const storyProps = {
  priceBreakdown: [
    {
      text: 'Base Price',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 500
      }
    },
    {
      text: 'Exterior Additions',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 300
      }
    },
    {
      text: 'Interior Additions',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 50000
      }
    },
    {
      text: 'Total cost',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 35500
      }
    }
  ]
};

const storyProps2 = {
  priceBreakdown: [
    {
      text: 'Base Price',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 500
      }
    },
    {
      text: 'Exterior Additions',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 300
      },
      helpText: 'Helpful info about Exterior Additions'
    },
    {
      text: 'Duration',
      value: '24 months',
      helpText: 'Lease duration'
    }
  ]
};

export const normal = args => <PriceTable {...args} />;
normal.args = storyProps;

export const withHelpText = args => <PriceTable {...args} />;
withHelpText.args = storyProps2;
