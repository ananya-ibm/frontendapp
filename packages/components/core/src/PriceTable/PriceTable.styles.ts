/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled, { css } from 'styled-components';
import { ifProp } from '@exo/frontend-common-style-utils';

export const PriceTable = styled('div')<{ hasHelpText?: boolean }>`
  ${props => ifProp(props, 'hasHelpText').then(css`
    padding-right: 2rem;
  `)}
`;

export const Row = styled('div')`
  border-bottom: 0.0625rem solid ${props => props.theme.colors.backgrounds.panels.primary.base};
  display: flex;
  justify-content: space-between;
  position: relative;
`;

export const Text = styled('div')`
  padding: 1rem 0;
`;

export const Price = styled('div')`
  border-bottom: 0;
  font-weight: 700;
  padding: 1rem 0;
`;

export const Help = styled('div')`
  display: flex;
  height: 100%;
  position: absolute;
  right: -2rem;
  top: 0;
  width: 2rem;

  & svg {
    margin: auto;
    max-width: 1rem;
  }
`;
