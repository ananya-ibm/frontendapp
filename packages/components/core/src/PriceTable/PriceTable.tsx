/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { InformationFilled } from '@carbon/react/icons';
import { MonetaryAmount } from '../MonetaryAmount/MonetaryAmount';
import * as S from './PriceTable.styles';

export const PriceTable = ({ priceBreakdown }: Props) => {
  const hasHelpText = Boolean(priceBreakdown.find(item => !!item.helpText));

  return (
    <S.PriceTable hasHelpText={hasHelpText}>
      {priceBreakdown.map(option => (
        <S.Row key={option.text}>
          {option.text && <S.Text>{option.text}</S.Text>}
          {option.amount && (
            <S.Price>
              <MonetaryAmount
                prefix={option.amount.prefix}
                currency={option.amount.currency}
                value={parseFloat(option.amount.value?.toString() ?? '0').toFixed(2)}
              />
            </S.Price>
          )}
          {option.value && <S.Price>{option.value}</S.Price>}
          {option.helpText && (
            <S.Help title={option.helpText}>
              <InformationFilled size={32} />
              <span className="sr">{option.helpText}</span>
            </S.Help>
          )}
        </S.Row>
      ))}
    </S.PriceTable>
  );
};

type Props = {
  priceBreakdown: {
    text?: string;
    amount: {
      prefix?: string;
      currency?: string;
      value: string | number;
    };
    value?: string | number;
    helpText?: string;
  }[];
};
