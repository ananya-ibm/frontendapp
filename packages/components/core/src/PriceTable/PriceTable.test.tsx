/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { renderWithTheme as render } from '@testUtils';
import { axe } from 'jest-axe';
import { PriceTable } from './PriceTable';

const testProps = {
  priceBreakdown: [
    {
      text: 'Base Price',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: '500'
      },
      helpText: 'Helpful info about Base Price'
    },
    {
      text: 'Exterior Additions',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: '300'
      }
    },
    {
      text: 'Interior Additions',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: '50000'
      }
    },
    {
      text: 'Total cost',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: '35500'
      }
    }
  ]
};

describe('<PriceTable /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<PriceTable {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });
  test('it passes an axe accesibility check', async () => {
    const { container } = render(<PriceTable {...testProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
