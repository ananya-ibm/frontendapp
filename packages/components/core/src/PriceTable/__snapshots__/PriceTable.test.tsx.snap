// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceTable /> component its snapshot matches 1`] = `
.c4 .price-list {
  color: rgba(22,22,22,0.25);
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c0 {
  padding-right: 2rem;
}

.c1 {
  border-bottom: 0.0625rem solid #f4f4f4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.c2 {
  padding: 1rem 0;
}

.c3 {
  border-bottom: 0;
  font-weight: 700;
  padding: 1rem 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  position: absolute;
  right: -2rem;
  top: 0;
  width: 2rem;
}

.c5 svg {
  margin: auto;
  max-width: 1rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Base Price
    </div>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        <span
          class="prefix"
        >
          +
        </span>
        <span
          class="value"
        >
          £500
        </span>
      </span>
    </div>
    <div
      class="c5"
      title="Helpful info about Base Price"
    >
      <svg
        aria-label="icon"
        fill="currentColor"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        size="32"
        viewBox="0 0 32 32"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      />
      <span
        class="sr"
      >
        Helpful info about Base Price
      </span>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Exterior Additions
    </div>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        <span
          class="prefix"
        >
          +
        </span>
        <span
          class="value"
        >
          £300
        </span>
      </span>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Interior Additions
    </div>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        <span
          class="prefix"
        >
          +
        </span>
        <span
          class="value"
        >
          £50,000
        </span>
      </span>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Total cost
    </div>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        <span
          class="prefix"
        >
          +
        </span>
        <span
          class="value"
        >
          £35,500
        </span>
      </span>
    </div>
  </div>
</div>
`;
